infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/batch/TransferBatchResponse.java

```diff
@@ -0,0 +1,52 @@
+package org.corfudb.infrastructure.log.statetransfer.batch;
+
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.corfudb.infrastructure.log.statetransfer.exceptions.StateTransferBatchProcessorException;
+
+import java.util.Optional;
+
+import static lombok.EqualsAndHashCode.*;
+import static org.corfudb.infrastructure.log.statetransfer.batch.TransferBatchResponse.TransferStatus.SUCCEEDED;
+
+/**
+ * A result of a transfer of a batch of addresses.
+ * If completed successfully, contains a status SUCCEEDED,
+ * a {@link #transferBatchRequest} that contains a list of addresses transferred,
+ * and an optional destination server.
+ * If completed exceptionally, contains a status FAILED, and a cause of failure,
+ * a {@link #transferBatchRequest} that contains an empty list,
+ * and an optional destination server.
+ */
+@Builder
+@Getter
+@EqualsAndHashCode
+public class TransferBatchResponse {
+
+    public enum TransferStatus {
+        SUCCEEDED,
+        FAILED
+    }
+
+    /**
+     * An original request.
+     */
+    @Default
+    @Getter
+    private final TransferBatchRequest transferBatchRequest = TransferBatchRequest.builder().build();
```

Comment:
We definitely need for the next PR. These addresses can be rescheduled to run on the different nodes in case of failures, so it's important to preserve this info.